appyters/Clustergrammer_Appyter/clustergrammer.ipynb

```diff
@@ -0,0 +1,554 @@
+    "            print(f'Filtered {cat_axis} number {cat_index}')\n",
+    "        except:\n",
+    "            print(\"Error: Could not filter by category\")\n",
+    "    if method == 'filter_name':\n",
+    "        try:\n",
+    "            net.filter_names(axis=names_axis, names=names_names)\n",
+    "            print(f'Filtered the following {names_axis}s: {names_names}')\n",
+    "        except:\n",
+    "            print(\"Error: Could not filter by name\")'''\n",
+    "    if method == 'filter_sum':\n",
+    "        try:\n",
+    "            net.filter_sum(inst_rc=sum_axis, threshold=sum_threshold, take_abs=sum_take_abs)\n",
+    "            print(f'Filtered {sum_axis}s under {sum_threshold} threshold, {\"taking\" if sum_take_abs == True else \"not taking\"} absolute value')\n",
+    "        except:\n",
+    "            print(\"Error: Could not filter by sum\")\n",
+    "    if method == 'filter_thresh':\n",
+    "        try:\n",
+    "            net.filter_threshold(inst_rc=threshold_axis, threshold=threshold_threshold, num_occur=threshold_num_occur)\n",
+    "            print(f'Filtered for values with {threshold_num_occur}{\"s\" if threshold_num_occur > 1 else \"\"} occurrences above {threshold_threshold} threshold')\n",
+    "        except:\n",
+    "            print(\"Error: Could not filter by threshold\")\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Clustering"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    net.cluster(dist_type=cluster_dist_type, \n",
+    "            run_clustering=True, \n",
+    "            dendro=cluster_dendro, \n",
+    "            views=['N_row_sum', 'N_row_var'], \n",
+    "            linkage_type='average', \n",
+    "            sim_mat=False, \n",
+    "            filter_sim=0.1, \n",
+    "            calc_cat_pval=False, \n",
+    "            run_enrichr=None, \n",
+    "            enrichrgram=None)\n",
+    "    print(f'Clustering preferences selected: {cluster_dist_type} distance, {\"loading\" if cluster_dendro == True else \"not loading\"} dendrogram')\n",
+    "except:\n",
+    "    print('ERROR: net.cluster(...) method did not run successfully')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate and display Clustergram"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clustergrammer_url = 'http://amp.pharm.mssm.edu/clustergrammer/matrix_upload/'\n",
+    "clustergrammer_df = net.export_df()\n",
+    "clustergrammer_fn = 'clustergrammer_fn.txt'\n",
+    "clustergrammer_df.to_csv(clustergrammer_fn, sep = '\\t')\n",
+    "r = requests.post(clustergrammer_url, files = {'file': open(clustergrammer_fn, 'rb')}).text\n",
```

Comment:
If you're importing clustergrammer and processing the data here anyway I'm a little confused about why you need to submit this to the clustergrammer website? Instead all necessary information is already here in the appyter where it can be computed and displayed.  If it's just submitting to clustergrammer than being an appyter is a bit pointless since clustergrammer can already do this.